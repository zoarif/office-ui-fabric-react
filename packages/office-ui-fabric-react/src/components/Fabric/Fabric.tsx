import * as React from 'react';
import {
  BaseComponent,
  createRef,
  customizable,
  getNativeProps,
  divProperties,
  classNamesFunction,
  getWindow,
  isDirectionalKeyCode
} from '../../Utilities';
import { getStyles } from './Fabric.styles';
import { IFabricProps, IFabricStyleProps, IFabricStyles } from './Fabric.types';

const getClassNames = classNamesFunction<IFabricStyleProps, IFabricStyles>();

@customizable('Fabric', ['theme'])
export class Fabric extends BaseComponent<IFabricProps, {
  isFocusVisible: boolean;
}> {
  private _rootElement = createRef<HTMLDivElement>();

  constructor(props: IFabricProps) {
    super(props);
    this.state = { isFocusVisible: false };
  }

  public render() {
    const classNames = getClassNames(getStyles,
      {
        ...this.props as IFabricStyleProps,
        ...this.state
      });
    const divProps = getNativeProps(this.props, divProperties);

    return (
      <div
        { ...divProps }
        className={ classNames.root }
        ref={ this._rootElement }
      />
    );
  }

  public componentDidMount(): void {
    const win = getWindow(this._rootElement.value);

    if (win) {
      this._events.on(win, 'mousedown', this._onMouseDown, true);
      this._events.on(win, 'keydown', this._onKeyDown, true);
    }
  }

  private _onMouseDown = (ev: MouseEvent): void => {
    this.setState({ isFocusVisible: false });
  }

  private _onKeyDown = (ev: KeyboardEvent): void => {
    if (isDirectionalKeyCode(ev.which)) {
      this.setState({ isFocusVisible: true });
    }
  }
}
