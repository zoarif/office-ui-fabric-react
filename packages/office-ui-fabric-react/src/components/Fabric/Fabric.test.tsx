import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Fabric } from './Fabric';

describe('Fabric', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <Fabric>test</Fabric>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
