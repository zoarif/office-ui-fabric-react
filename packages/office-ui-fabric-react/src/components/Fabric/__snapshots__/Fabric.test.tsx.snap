// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fabric renders correctly 1`] = `
<div
  className=
      ms-Fabric
      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        color: #333333;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 14px;
        font-weight: 400;
      }
      & button {
        font-family: inherit;
      }
      & input {
        font-family: inherit;
      }
      & textarea {
        font-family: inherit;
      }
      button {
        overflow: visible;
      }
>
  test
</div>
`;
